src/main/java/me/fallenbreath/yamlang/YamlangPlugin.java

```diff
@@ -19,13 +21,24 @@ public void apply(Project project)
 			{
 				String taskName = String.format("yamlangConvert%sResources", StringUtils.capitalize(sourceSet.getName()));
 				project.getLogger().info("Hooking sourceset {}", sourceSet);
-				Task processResources = project.getTasks().getByName(sourceSet.getProcessResourcesTaskName());
 
-				project.getTasks().register(taskName, YamlangConvertor.class, task -> {
-					task.setSourceSet(sourceSet);
-					task.onlyIf(s -> !processResources.getState().getUpToDate());
+				TaskProvider<YamlangConvertResourcesTask> provider = project.getTasks().register(taskName, YamlangConvertResourcesTask.class, task -> {
+					task.dependsOn(sourceSet.getProcessResourcesTaskName());
+					task.getOutputs().upToDateWhen(t -> false);
+
+					DirectoryProperty destination = project.getObjects().directoryProperty().fileValue(sourceSet.getOutput().getResourcesDir());
+					Provider<Directory> inputDir = destination.dir(extension.getInputDir().orElse(""));
+					Provider<Directory> outputDir = destination.dir(extension.getOutputDir().orElse(extension.getInputDir().orElse("")));
+
+					task.getInputDirectory().set(inputDir);
+					task.getOutputDirectory().set(outputDir);
+					task.getTargetFilePattern().set(extension.getTargetFilePattern());
+					task.getPreserveYaml().set(extension.getPreserveYaml());
+					task.getCharset().set(extension.getCharset());
+					task.getOwolibRichTranslations().set(extension.getOwolibRichTranslations());
 				});
-				processResources.finalizedBy(taskName);
+				project.getTasks().getByName(sourceSet.getProcessResourcesTaskName(), task -> task.finalizedBy(provider));
+				project.getTasks().getByName(sourceSet.getClassesTaskName(), task -> task.dependsOn(provider));
```

Comment:
What is this dependency with `classes` task for?

Reply:
The `classes` task implicitly depends on our task by depending on the output resources, which we modify. Gradle requires that that all dependencies between tasks are explicit.